src/test/test_pidfile.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash -e
+
+#
+# Generic read/write from object store test
+#
+
+# Includes
+source ../qa/workunits/ceph-helpers.sh
+
+function run() {
+    local dir=$1
+    shift
+
+    export CEPH_MON="127.0.0.1:17108"
+    export CEPH_ARGS
+    CEPH_ARGS+="--fsid=$(uuidgen) --auth-supported=none "
+    CEPH_ARGS+="--mon-host=$CEPH_MON "
+
+    local funcs=${@:-$(set | sed -n -e 's/^\(TEST_[0-9a-z_]*\) .*/\1/p')}
+    for func in $funcs ; do
+        $func $dir || return 1
+    done
+}
+
+
+function TEST_normal() {
+    local dir=$1
+    setup $dir 
+    run_mon $dir a
+    run_osd $dir 0
+    teardown $dir 
+}
+
+function TEST_without_pidfile() {
+   local dir=$1
+   setup $dir
+   RUNID=`uuidgen`
+   run_mon $dir a --pid-file= --daemonize=$RUNID || { teardown_unexist_pidfile $dir; return 1; } 
+   run_osd $dir 0 --pid-file= --daemonize=$RUNID || { teardown_unexist_pidfile $dir; return 1; }
+   teardown_unexist_pidfile $dir || return 1
+}
+
+function teardown_unexist_pidfile() {
+   local dir=$1
+   local delays=('0' '1' '3' '5' '10' '10' '20' '60')
+   pids=$(ps aux|awk "/cep[h].*$RUNID.*/ {print \$2}")
+   for i in $pids
+   do
+     local kill_complete="0"
+     count=0
+     while [ $kill_complete -eq "0" ]
```

Comment:
you break at line 59 when `kill_complete` is "1", so this line is not necessary at all?